eng/Versions.props

```diff
@@ -179,7 +179,7 @@
     <!-- ICU -->
     <MicrosoftNETCoreRuntimeICUTransportVersion>7.0.0-rtm.24115.1</MicrosoftNETCoreRuntimeICUTransportVersion>
     <!-- MsQuic -->
-    <MicrosoftNativeQuicMsQuicVersion>2.2.3</MicrosoftNativeQuicMsQuicVersion>
+    <MicrosoftNativeQuicMsQuicVersion>2.3.5</MicrosoftNativeQuicMsQuicVersion>
```

Comment:
New failure states the package is not found in the dotnet feeds (we don't search in nuget):
```
src/libraries/System.Net.Quic/src/System.Net.Quic.csproj(0,0): error NU1102: (NETCORE_ENGINEERING_TELEMETRY=Restore) Unable to find package Microsoft.Native.Quic.MsQuic.Schannel with version (>= 2.3.5)
  - Found 17 version(s) in dotnet-public [ Nearest version: 2.3.1 ]
  - Found 1 version(s) in dotnet7-transport [ Nearest version: 2.2.3-ci.391104 ]
  - Found 0 version(s) in richnav
  - Found 0 version(s) in darc-pub-dotnet-emsdk-c7ba18b
  - Found 0 version(s) in dotnet-eng
  - Found 0 version(s) in dotnet-libraries
  - Found 0 version(s) in dotnet-tools
  - Found 0 version(s) in dotnet7
  ```
  
  I think I just need to push it to the transport feed, then restart the CI.